### PR TITLE
CU-86byatqw4_Create-Load-Configuration-Endpoint

### PR DESCRIPTION
**Pull request by harmochiky**
**Changes**
Added GET /configuration endpoint to fetch configurations

**Testing**
To test, paste a `config-reference.json` in the `devops/linux/docker/docker_data/data-apps/api/conf` folder then call the GET /configuration endpoint. 

## Summary by CodeRabbit

- **New Features**
  - Added new API configuration constants and parameters enhancing the configuration process.
  - Introduced nested record classes for detailed configuration management in models.
- **Refactor**
  - Improved readability and alignment in various Java classes without altering existing logic.
- **Documentation**
  - Updated method signatures and variable declarations for better clarity.
- **Chores**
  - Adjusted Docker environment configurations and script exports for API settings.
- **Style**
  - Standardized quoting styles and added indentation in Docker YAML files for better readability.

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86byatqw4 Create Load Configuration Endpoint](https://app.clickup.com/t/86byatqw4)

**Comment by rcrichton** [comment]
Task linked: [CU-86by1w64p JeMPI UI Configuration](https://app.clickup.com/t/86by1w64p)

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/BackEnd.java [hunk 1]
```diff
@@ -473,6 +474,21 @@ private Behavior<Event> getSqlDashboardDataHandler(final SQLDashboardDataRequest
       return Behaviors.same();
    }
 
+   private Behavior<Event> getConfigurationHandler(final GetConfigurationRequest request) {
+      String currentPath = System.getProperty("user.dir");
+      String configJsonFilePath = currentPath + "/conf/config-reference.json";
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/BackEnd.java:479]: @harmochiky this needs to be updated and set in the app config as an environment variable

File: JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/BackEnd.java [hunk 2]
```diff
@@ -537,6 +553,12 @@ public record GetGidsAllRequest(ActorRef<GetGidsAllResponse> replyTo) implements
    public record GetGidsAllResponse(List<String> records) implements EventResponse {
    }
 
+   public record GetConfigurationRequest(ActorRef<GetConfigurationResponse> replyTo) implements Event {
+   }
+
+   public record GetConfigurationResponse(String configuration) implements EventResponse {
+   }
+
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibAPI/src/main/java/org/jembi/jempi/libapi/BackEnd.java:561]: @harmochiky this response type and its properties need to be more explicit than bundling it all into a string. With the current solution the file dictates the format of the response. What happens when the file format is incorrectly saved?

**Review by MatthewErispe — CHANGES_REQUESTED** [review]

**Comment by harmochiky** [comment]
PR ready for you @MatthewErispe

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        public record DeterministicLink(
+                        @JsonProperty("QUERY_LINK_DETERMINISTIC_A") QueryLink queryLinkDeterministicA,
+                        @JsonProperty("QUERY_LINK_DETERMINISTIC_B") QueryLink queryLinkDeterministicB) {
+        }
+
+        public record ProbabilisticLink(
+                        @JsonProperty("QUERY_LINK_PROBABILISTIC") QueryLink queryLinkProbabilistic) {
+        }
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java:69]: @harmochiky use a queryLink list instead

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        public record LinkMetaData(
+                        String comparison,
+                        List<Double> comparisonLevels,
+                        double m,
+                        double u) {
+        }
+
+        public record Rules(
+                        LinkRules link) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java:54]: @harmochiky please add matchRules and ValidateRules

**Review by MatthewErispe — CHANGES_REQUESTED** [review]
@harmochiky I am happy with your changes made apart from a few more

**Comment by MatthewErispe** [comment]
@coderabbitai review

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java (hunk 3 above)
Comment (MatthewErispe) [reply JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java:69]: Including these in additional changes made to configuration

File: JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java (hunk 4 above)
Comment (MatthewErispe) [reply JeMPI_Apps/JeMPI_LibShared/src/main/java/org/jembi/jempi/shared/models/ConfigurationModel.java:54]: Including this request in new task for additional changes made to configuration

**Review by MatthewErispe — APPROVED** [review]
@harmochiky happy with your additional changes. Added a few refactor changes suggested by coderabbitai

**Pull request merged**